src/bin/src/systems/keep_alive_system.rs

```diff
@@ -84,10 +85,11 @@ impl System for KeepAliveSystem {
             if let Err(e) = state.broadcast(&packet, broadcast_opts).await {
                 error!("Error sending keep alive packet: {}", e);
             };
-
+            // TODO, this should be configurable as some people may have bad network so the clients may end up disconnecting from the server moments before the keep alive is sent
```

Comment:
15 seconds delay means the client gets 2 chances. The default timeout for keepalive is 30 seconds.

Reply:
Ok I have no issue changing this since we will be making it configurable later, in the meantime though, there's an issue where the keep alive runs twice every 30 seconds or 15, whatever the timeout is configured to be, can you look into it, I don't really get why its doing this, everything should be done synchronously in the loop so I'm very lost.